src/main/java/graphql/execution/instrumentation/dataloader/FieldLevelTrackingApproach.java

```diff
@@ -29,60 +32,63 @@
 
     private static class CallStack implements InstrumentationState {
 
-        private final LevelMap expectedFetchCountPerLevel = new LevelMap();
-        private final LevelMap fetchCountPerLevel = new LevelMap();
-        private final LevelMap expectedStrategyCallsPerLevel = new LevelMap();
-        private final LevelMap happenedStrategyCallsPerLevel = new LevelMap();
-        private final LevelMap happenedOnFieldValueCallsPerLevel = new LevelMap();
+        private final Map<Integer, Integer> expectedFetchCountPerLevel = new LinkedHashMap<>();
+        private final Map<Integer, Integer> fetchCountPerLevel = new LinkedHashMap<>();
+        private final Map<Integer, Integer> expectedStrategyCallsPerLevel = new LinkedHashMap<>();
+        private final Map<Integer, Integer> happenedStrategyCallsPerLevel = new LinkedHashMap<>();
+        private final Map<Integer, Integer> happenedOnFieldValueCallsPerLevel = new LinkedHashMap<>();
+
```

Comment:
Why did you change this.  Our custom LevelMap (array of ints) surely is more memory efficeient than a LinkedHashMap

Reply:
Apologies, we forked the repo internally to analyse and test these changes and when I have tried to move this commit out into the public repo for you to see I have somehow pulled in some unrelated things (that to be honest I haven't seen before), let me fix this PR.